src/Compilers/CSharp/Portable/Parser/SyntaxParser.cs

```diff
@@ -757,32 +746,20 @@ protected virtual TNode WithAdditionalDiagnostics<TNode>(TNode node, params Diag
             }
         }
 
-        protected TNode AddError<TNode>(TNode node, ErrorCode code) where TNode : GreenNode
-        {
-            return AddError(node, code, Array.Empty<object>());
-        }
-
-        protected TNode AddErrorAsWarning<TNode>(TNode node, ErrorCode code, params object[] args) where TNode : GreenNode
-        {
-            Debug.Assert(!node.IsMissing);
-            return AddError(node, ErrorCode.WRN_ErrorOverride, MakeError(node, code, args), (int)code);
-        }
-
-        protected TNode AddError<TNode>(TNode nodeOrToken, ErrorCode code, params object[] args) where TNode : GreenNode
+        /// <summary>
+        /// Given a "missing" node or token (one where <see cref="GreenNode.IsMissing"/> must be true, determines the
+        /// ideal location to place the diagnostic for it.  The intuition here is that we want to place the diagnostic
+        /// on the token that "follows" this 'missing' entity if they're on the same line.  Or, place it at the end of
+        /// the 'preceding' token if the following token is on the next line.
+        /// </summary>
+        protected (int offset, int width) GetDiagnosticSpanForMissingNodeOrToken(GreenNode missingNodeOrToken)
```

Comment:
the major change here is that this now should be the common location taht all diags for missing constructs go through.  previously there were two helpers and it was possible for callers to call into either.